firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistributionLifecycleNotifier.java

```diff
@@ -80,6 +91,25 @@ Activity getCurrentActivity() {
     }
   }
 
+  // TODO (mannyjimenez) Put a lock on this in the future.
+  Task<Activity> getForegroundActivity() {
+    if (currentActivity != null) {
+      return Tasks.forResult(currentActivity);
+    }
+    TaskCompletionSource<Activity> task = new TaskCompletionSource<>();
+
+    addOnActivityResumedListener(
+        new OnActivityResumedListener() {
+          @Override
+          public void onResumed(Activity activity) {
+            task.setResult(activity);
+            removeOnActivityResumedListener(this);
+          }
+        });
+
+    return task.getTask();
+  }
+
   Activity getNonNullCurrentActivity() throws FirebaseAppDistributionException {
```

Comment:
Do we still need this method? Could we remove the `APP_BACKGROUNDED` and `FOREGROUND_ACTIVITY_NOT_AVAILABLE` methods?

Reply:
removed the method and error codes.